Reverse sentence

shrutivanw opened the pull request:
CS Fundamentals' assignment to reverse the words in a sentence in place while preserving the count of white spaces.

shrutivanw committed:
Reverse sentence

CS Fundamentals' assignment to reverse the words in a sentence in place while preserving the count of white spaces.

Hamled commented on README.md:
```diff
@@ -1 +1,9 @@
+* Design and implement a method to reverse the words in a sentence *in place*. For example, if the method is called with input parameter of "Yoda&nbsp;&nbsp;&nbsp;is&nbsp;&nbsp;&nbsp;&nbsp;awesome", the method should update the input string object to have the value "awesome&nbsp;&nbsp;&nbsp;&nbsp;is&nbsp;&nbsp;&nbsp;Yoda". Note that the count of white spaces between words is preserved.
+* Share and explain the time and space complexities for your solution.
+* At minimum, your implementation should pass the basic tests.
+**Note**: Do not use Ruby provided functionality for `.reverse` and `.reverse!`. You may use `.length`.
```
We should have an extra blank line above this so that the note goes into its own paragraph.

Hamled commented on lib/reverse_sentence.rb:
```diff
@@ -0,0 +1,4 @@
+# A method to reverse the words in a sentence, in place.
+def reverse_sentence(my_sentence)
+  my_sentence << "NOT IMPLEMENTED"
```
See [my comment](https://github.com/AdaGold/reverse_words/pull/1/files#r160554004) on the `reverse_words` repo, which applies equally here.

Hamled commented on specs/reverse_sentence_spec.rb:
```diff
@@ -0,0 +1,92 @@
+
+describe "reverse sentence" do
+  describe "basic tests" do
+    describe "test 1" do
```
Same as the other repos, we should probably remove the `describe` blocks that are immediate parents of individual tests.

Hamled commented on specs/reverse_sentence_spec.rb:
```diff
@@ -0,0 +1,92 @@
+
+  # if it's a string parameter, check for empty
+  describe "empty string input test" do
+    it "reverse an empty sentence" do
```
I think this empty string test and the nil test below it could be moved into the edge cases section, and a few of those tests (such as sentence with multiple words) could be moved into the basic tests section.

shrutivanw committed:
Incorporating good feedback

- Changed the template method definition to raise NotImplementedError
- Moved nil and empty test cases to edge cases
- Removed inner most describe from individual tests in basic and edge case tests
- Updated readme file to show the newline before Notes.

shrutivanw commented:
Thank you @Hamled PR updated to incorporate feedback.
